### ISSUE TITLE
Make tabs tabbable again! (skip setting tabindex to -1)

### ISSUE THREAD

**Issue by limonte** [vaadin/vaadin-tabs#97]
Currently, `vaadin-tabs` aren't tabbable. Sounds funny, yeah :)

We've implemented the unique behavior in keyboard a11y, even our own demos are inconsistent, tabbing works nicely in `<nav id="nav">` but not in `vaadin-tabs` examples:

![peek 2018-05-07 12-20](https://user-images.githubusercontent.com/6059356/39694772-15c044c2-51f1-11e8-9534-8219de5eb651.gif)

**Expected**: ability to focus "Tab two" with <kbd>Tab</kbd>

**Actual**: inability to focus "Tab two" with <kbd>Tab</kbd>

Our users are asking about workaround and how to set `tabindex` of `vaadin-tab`s to `0`.

I think we should provide the similar experience as most of tabs plugins provide, namely make tabs tabbable, namely by pressing <kbd>Tab</kbd> users will be able to go through all tabs.

The current behavior could be optional, users will be able to turn it on by setting some property (name to be decided)

**Comment by jouni** [comment]
According to [WAI-ARIA recommendations](https://www.w3.org/TR/wai-aria-practices-1.1/#tabpanel):

For the tab list:
- Tab: When focus moves into the tab list, places focus on the active tab element . When the tab list contains the focus, moves focus to the next element in the page tab sequence outside the tablist, which is typically either the first focusable element inside the tab panel or the tab panel itself.
- When focus is on a tab element in a horizontal tab list:
  - Left Arrow: moves focus to the previous tab. If focus is on the first tab, moves focus to the last tab. Optionally, activates the newly focused tab (See note below).
  - Right Arrow: Moves focus to the next tab. If focus is on the last tab element, moves focus to the first tab. Optionally, activates the newly focused tab (See note below)

This is specifally so that users can quickly skip over a long tab list to get to the tab content.

**Comment by tomivirkki** [comment]
> We've implemented the unique behavior in keyboard a11y

Here's another example with the same behavior: https://www.webcomponents.org/element/PolymerElements/paper-tabs/demo/demo/index.html

**Comment by platosha** [comment]
An issue with tabbing through individual tabs is that once there are many tabs, it might get really frustrating to tab through them all to navigate to the next component.

So agree with @jouni, this is a nice feature of `<vaadin-tabs>` that allows users to skip over the entire tabs widget, no matter how many tabs are there.

**Comment by limonte** [comment]
> This is specifically so that users can quickly skip over a long tab list to get to the tab content.

> So agree with @jouni, this is a nice feature of <vaadin-tabs> that allows users to skip over the entire tabs widget, no matter how many tabs are there.

One of the [essential skills](https://uxplanet.org/mvs-8e15cf2ca0e) of good UX designers is to know existed solutions. Please take a look at GitHub UI, there are tabbable tabs and "Skip to content" button.

Once again, I'm proposing adding a new **optional** feature, which could be added to the next minor version if we're not making the new behavior as a default.

**Comment by jouni** [comment]
Github tabs are essentially links, they cause a full page reload most of the time. And yes, traditionally web page tabs are implemented with links and hence are in the tab order.

The use case for vaadin-tabs is more on the full-blown application side, where my experience is that tabs are navigated with arrow keys and not the tab key.

Options are fine, and it’s the maintenance team’s call if they want to take on the added cost of maintaining more options. Personally, I’m not seeing enough need for such an option at the moment. If we would decide to add an option, I would keep the current behavior the default one, so we don’t have to make a new major version (might make sense to remove the “requires new major” label as we have an option to not require it).

My suggestion, for now, would be to implement such tabs as custom using `<a>` elements, if you need “tabbable tabs”.

**Comment by limonte** [comment]
> my experience is that tabs are navigated with arrow keys and not the tab key.
> I don’t have access to any these kinds of apps that I referred

You have an experience with apps you don't have access to. Sounds contradictory :upside_down_face: 

> I can’t speak for screen reader users (and I don’t have much experience using apps with keyboard-only), but I think my preference would be one tab-stop per tab bar if there are more than a handful of tabs.

The easiest way to check out what is the best keyboard a11y, it's quite logical to test one of a11y-related projects, e.g.

- https://www.usability.gov/what-and-why/accessibility.html
- https://a11yproject.com/ 
 
> vaadin-tabs is designed to support the latter case (a large number of tabs).

Isn't a large number of tabs the poor UX by itself? In my opinion, it is. The reason is simple. As a user I either can't see all tabs:

![image](https://user-images.githubusercontent.com/6059356/39743775-b75081bc-52aa-11e8-99a7-33d8581f3ae9.png)

or see an extremely wide app: 

![image](https://user-images.githubusercontent.com/6059356/39743805-d2a35a0c-52aa-11e8-9b2e-6eb8ed856692.png)

But please do not make a desicion based on 1 person opinion and explore related questions/answers https://ux.stackexchange.com/search?q=many+tabs to see different opinions.

**Comment by tomivirkki** [comment]
Then there is this https://developer.mozilla.org/en-US/docs/Web/Accessibility/Keyboard-navigable_JavaScript_widgets#Grouping_controls

**Comment by limonte** [comment]
> For the guidelines they promote, at least a11yproject has a “Simple Tabs” example that opts for arrow navigation: https://codepen.io/svinkle/pen/edmDF

Please pay attention to this `span` from their codepen:

```html
<span id="tab-widget-description" class="visuallyhidden">Use left and right arrows to navigate between tabs.</span>
```

which will be read by screen readers. We don't have anything like that.

---

> This is specifically so that users can quickly skip over a long tab list to get to the tab content.

> An issue with tabbing through individual tabs is that once there are many tabs, it might get really frustrating to tab through them all to navigate to the next component.

This should be solved in a completely different way, which is the WCAG 2.0 (A) MUST: https://dequeuniversity.com/rules/axe/2.2/bypass?application=lighthouse

![image](https://user-images.githubusercontent.com/6059356/39913900-5ab1b980-553e-11e8-9f5f-145cf2739723.png)
